.github/workflows/ci.yml

```diff
@@ -79,18 +79,10 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          python -m pip install --upgrade tox tox-gh-actions
```

Comment:
@WisdomPill You may want to think about getting these out of your github actions and into something like a dev_dependencies (or pyproject.toml dev area). This way they're not as hidden from users.

Reply:
Yes, I guess I will add some kind of dev script to setup the environment.

I still do not really understand how tox installs dependencies and where is it getting `redislatest`.

These are my first steps with tox

Reply:
Got it. Here's how I think about tox with docker:

- Define a series of docker images that may or may not get used
- Define tox environments that depend on individual dockers
- Add pre_commands or dependencies if/as necessary
- Run tests

For [one project](https://github.com/RedisGears/rgsync/blob/master/tox.ini) I use this pattern to parameterize a pytest option, so that we can control test execution and dockers being run. It may be a better example for your pattern.